dashboard/test/ui/step_definitions/steps.rb

```diff
@@ -271,6 +271,23 @@ def replace_hostname(url)
   @browser.execute_script %q-sessionStorage.setItem('progress', '{"hourofcode":{"' + appOptions.serverLevelId + '":100}}')-
 end
 
+Then /^I verify progress in the header of the current page is "([^"]*)" for level (\d+)/ do |test_result, level|
+  steps %{
+    And I wait to see ".progress_container"
+    And I wait for 10 seconds
+    And element ".progress_container a.level_link:nth(#{level.to_i - 1})" has class "#{test_result}"
+  }
+end
+
+Then /^I navigate to the course page and verify progress for course "([^"]*)" stage (\d+) level (\d+) is "([^"]*)"/ do |course, stage, level, test_result|
+  steps %{
+    Then I am on "http://studio.code.org/s/#{course}"
+    And I wait to see ".user-stats-block"
+    And I wait for 10 seconds
```

Comment:
Allow time for the AJAX request to return in scenarios where progress is loaded async.